### PR TITLE
CLOUDP-163493 - ci/cd cleanup

### PR DESCRIPTION
**Pull request by wtrocki**
## Description

Changes

- [x] Require SDK changes with each ./tools PRs. those might be breaking SDK or introducing an invalid open API file)
- [x] Run lint and unit tests for tooling changes (as we did them to validate the correctness of the transformations) and templates
- [x] Minor cleanup - removing redundant checks from the Open API validator path
   - linting - generally templates proving our ability for passing lint errors,
   - compatibility tests - used for tooling but will be breaking/blocking jobs when a new version of that API is introduced. 
- [x] README updates explaining those changes

### CODE REVIEW

**Review by gssbzn — COMMENTED** [review]
should https://github.com/mongodb/go-client-mongodb-atlas/actions/runs/4284032808/jobs/7460419036 be passing?

File: .github/workflows/tools.yaml [hunk 1]
```diff
@@ -18,3 +18,16 @@ jobs:
...
+      - name: Fail when generate introducing diff
+        run: |
+          git diff --exit-code
+      - name: Make sure that no breaking changes were introduced
+        run: | 
+          make tools v2-lint v2-test
+
+
+
+
```
Comment (gssbzn) [inline .github/workflows/tools.yaml:33]: super nit but let's remove the extra empty lines
```suggestion
      - name: Make sure that no breaking changes were introduced
        run: | 
          make tools v2-lint v2-test
```

File: api/v1alpha/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Go API client
+# Go API client
```
Comment (gssbzn) [inline api/v1alpha/README.md:1]:
```suggestion
# Go API client
```

File: GNUmakefile [hunk 3]
```diff
@@ -53,14 +53,12 @@ check-version:
...
 .PHONY: openapi-pipeline
-openapi-pipeline: tools
+openapi-pipeline:
 	echo "Running OpenAPI Generation and Validation process"
 	$(MAKE) -C tools clean_client
 	echo "Running client generation"
 	$(MAKE) -C tools generate_client
 	echo "Validating generated SDK"
-	$(MAKE) v2-lint
-	$(MAKE) v2-test
```
Comment (gssbzn) [inline GNUmakefile:63]: to follow some common patterns we've done across repos let's add a new target `v2-check` which is `v2-test v2-lint` somehwat strong opinion on that order to follow a red->green->refactor approach

File: .github/workflows/tools.yaml [hunk 4]
```diff
@@ -18,3 +18,16 @@ jobs:
...
           npm run ci
+      - working-directory: ./tools
+        run: |
+          make clean_and_generate
+      - name: Fail when generate introducing diff
+        run: |
+          git diff --exit-code
+      - name: Make sure that no breaking changes were introduced
+        run: | 
+          make tools v2-lint v2-test
```
Comment (gssbzn) [inline .github/workflows/tools.yaml:29]: [q] any reason to use `v2-lint` instead of the GH action for golangci-lint?

File: GNUmakefile (hunk 3 above)
Comment (wtrocki) [reply GNUmakefile:63]: sounds good

File: .github/workflows/tools.yaml (hunk 4 above)
Comment (wtrocki) [reply .github/workflows/tools.yaml:29]: Good point. Swapping

Comment (wtrocki) [reply .github/workflows/tools.yaml:29]: We do have global lint so this one sems redundant

File: GNUmakefile (hunk 3 above)
Comment (wtrocki) [reply GNUmakefile:63]: I'm removing those in favour of global test and lint

File: tools/README.md [hunk 5]
```diff
@@ -36,3 +36,19 @@ We use mustache comments to mark templates as modified:
 
 - `{{! X-XGEN-CUSTOM }}` - usually present at the top of the file - means that the whole file is currently customized.
 - `{{! X-XGEN-MODIFIED }}` - is put into existing templates to note areas in the code that were customized.
+
+#### Contributing
+
+1. After making change validate correctness of your changes by running
+
+```
+(cd .. && make v2-lint v2-test)
```
Comment (gssbzn) [inline tools/README.md:45]: seems outdated now

**Review by gssbzn — APPROVED** [review]
LGTM :shipit:

**Comment by wtrocki** [comment]
Awesome review!

**Pull request merged**